src/test/java/uk/gov/pay/products/resources/GatewayAccountResourceTest.java

```diff
@@ -130,4 +129,38 @@ public void shouldError_whenUpdatingAServiceName_withMandatoryFieldsMissing() th
                 .then()
                 .statusCode(400);
     }
+
+    @Test
+    public void shouldError_whenInvalidOperation() throws Exception {
+        ImmutableMap<Object, Object> payload = ImmutableMap.builder()
+                .put(FIELD_OPERATION, "remove")
+                .put(FIELD_OPERATION_PATH, PATH)
+                .put(FIELD_VALUE, VALUE)
+                .build();
+
+        givenAuthenticatedSetup()
+                .contentType(APPLICATION_JSON)
+                .accept(APPLICATION_JSON)
+                .body(mapper.writeValueAsString(payload))
+                .patch(format("/v1/api/gateway-account/%s", randomInt()))
+                .then()
+                .statusCode(400);
+    }
+
+    @Test
+    public void shouldError_whenInvalidPath() throws Exception {
+        ImmutableMap<Object, Object> payload = ImmutableMap.builder()
+                .put(FIELD_OPERATION, OP)
+                .put(FIELD_OPERATION_PATH, "gateway_id")
+                .put(FIELD_VALUE, VALUE)
+                .build();
+
+        givenAuthenticatedSetup()
+                .contentType(APPLICATION_JSON)
+                .accept(APPLICATION_JSON)
+                .body(mapper.writeValueAsString(payload))
+                .patch(format("/v1/api/gateway-account/%s", randomInt()))
+                .then()
+                .statusCode(400);
```

Comment:
As long as you have one test case for 400, the rest of the cases can just be unit tested. every integration test is expensive.